samples/table_insert_rows.py

```
@@ -12,23 +12,28 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-
-def table_insert_rows(table_id):
-
-    # [START bigquery_table_insert_rows]
-
-    from google.cloud import bigquery
-
-    # Construct a BigQuery client object.
-    client = bigquery.Client()
-
-    # TODO(developer): Set table_id to the ID of the model to fetch.
-    # table_id = "your-project.your_dataset.your_table"
-
-    table = client.get_table(table_id)  # Make an API request.
-    rows_to_insert = [(u"Phred Phlyntstone", 32), (u"Wylma Phlyntstone", 29)]
-
-    errors = client.insert_rows(table, rows_to_insert)  # Make an API request.
-    if errors == []:
-        print("New rows have been added.")
-    # [END bigquery_table_insert_rows]
+# [START bigquery_table_insert_rows]
+
+from google.cloud import bigquery
+
+# Construct a BigQuery client object.
+client = bigquery.Client()
+
+# TODO(developer): Set table_id to the ID of the model to fetch.
+table_id = "your-project.your_dataset.your_table"
+#Explicitly define the schema. (Optional)
+schema = [
+    bigquery.SchemaField("col_1", "STRING"),
+    bigquery.SchemaField("col_2", "INTEGER"),
+    ]
+rows_to_insert = [(u"Phred Phlyntstone", 32), (u"Wylma Phlyntstone", 29)] #populate data for entry
+
+try:
+    if 'schema' not in globals():
+        schema = client.get_table(table_id).schema  # Get table schema if none provided
+    client.insert_rows(table_id, selected_fields=schema, rows_to_insert)  # Stream data to BQ
+    print("New rows have been added.")
+except ValueError:
+    print("Table’s schema is not set or rows is not a Sequence.")
```

No need to catch this `ValueError`. Just let the exception bubble up.